File: src/main/java/gargoyle/netsquares/store/prop/PrefEnumProperty.java
package gargoyle.netsquares.store.prop;

public class PrefEnumProperty<E extends Enum<E>> extends BasePrefProperty<E> {
    public PrefEnumProperty(Class<?> pkg, String key, E defaultValue) {
        super(pkg, key, defaultValue);
    }

    @Override
    protected void _set(E value) {
        preferences.put(key, (value).name());
    }

    public E get(E defaultValue) {
        String name = preferences.get(key, null);
        return name == null ? defaultValue : Enum.valueOf(defaultValue.getDeclaringClass(), name);
    }
}

File: src/main/kotlin/gargoyle/netsquares/store/prop/PrefEnumProperty.kt
package gargoyle.netsquares.store.prop

import kotlin.reflect.KClass

class PrefEnumProperty<E : Enum<E>>(pkg: KClass<*>, key: String, defaultValue: E) :
    BasePrefProperty<E>(pkg, key, defaultValue) {
    override fun _set(value: E) {
        preferences.put(key, value.name)
    }

    @Suppress("UNCHECKED_CAST")
    override fun get(defaultValue: E): E {
        val name = preferences[key, null]
        return if (name == null) defaultValue
        else {
            java.lang.Enum.valueOf(defaultValue.javaClass.declaringClass as Class<E>, name)
        }
    }
}
